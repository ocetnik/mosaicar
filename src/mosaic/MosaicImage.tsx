import * as React from 'react';
import { RefObject } from 'react';

interface IMosaicImageProps {
    canvasRef: RefObject<HTMLCanvasElement>;
    mosaicSvgString: string;
}

class MosaicImage extends React.Component<IMosaicImageProps, {}> {
    public render() {
        return this.props.mosaicSvgString === ''
            ? <div>
                <canvas ref={this.props.canvasRef} />
            </div>
            : <div dangerouslySetInnerHTML={{ __html: this.props.mosaicSvgString }} />;
    }
}

export default MosaicImage;
