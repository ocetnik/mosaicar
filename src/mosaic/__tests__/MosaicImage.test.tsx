import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MosaicImage from '../MosaicImage';

describe('MosaicImage', () => {
    const ref = React.createRef<HTMLCanvasElement>();

    test('Should display canvas when mosaicSvgString === \'\'', () => {
        const component = renderer.create(
            <MosaicImage
                canvasRef={ref}
                mosaicSvgString={''}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Should display SVG string when mosaicSvgString !== \'\'', () => {
        const svgString = '<svg width="100" height="100"><circle cx="50" cy="50" r="40"' +
            ' stroke="green" stroke-width="4" fill="yellow" /></svg>';

        const component = renderer.create(
            <MosaicImage
                canvasRef={ref}
                mosaicSvgString={svgString}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
