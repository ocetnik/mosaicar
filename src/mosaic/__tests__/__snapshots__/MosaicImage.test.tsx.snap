// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MosaicImage Should display SVG string when mosaicSvgString !== '' 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"green\\" stroke-width=\\"4\\" fill=\\"yellow\\" /></svg>",
    }
  }
/>
`;

exports[`MosaicImage Should display canvas when mosaicSvgString === '' 1`] = `
<div>
  <canvas />
</div>
`;
